src/Spout/Writer/XLSX/Internal/Worksheet.php

```diff
@@ -133,10 +133,18 @@ public function addRow($dataRow, $style)
 
         $rowXML = '<row r="' . $rowIndex . '" spans="1:' . $numCells . '">';
 
-        foreach($dataRow as $cellValue) {
+        foreach($dataRow as $cell) {
+            if (is_array($cell)) {
```

Comment:
A parameter that can be of multiple types hides the complexity of the program. If you pass an number, it will do X but if you pass an array it will do Y. While it's fine to do different things, the expectation of what a function does should be clear. In this example, there should probably be 2 functions (one taking a number, one taking an array) with explicit names.

The problem is less the difference between a number and a string but more between a single element VS a collection. Those represent really different things and should be treated carefully.